rust_snuba/src/processors/generic_metrics.rs

```diff
@@ -240,6 +246,17 @@ where
     T: Parse + Serialize,
 {
     let payload_bytes = payload.payload().context("Expected payload")?;
+    let use_case: MessageUseCase = serde_json::from_slice(payload_bytes)?;
+    let killswitch_config = get_str_config("key");
+
+    if killswitch_config.is_ok() {
+        if let Some(str) = killswitch_config.unwrap() {
+            if use_case.use_case_id == str {
+                return Ok(InsertBatch::skip());
+            }
+        }
+    }
```

Comment:
Untested:

```suggestion
    if let Some(killswitch) = killswitch_config.ok().flatten() {
        let use_case: MessageUseCase = serde_json::from_slice(payload_bytes)?;
        if killswitch == use_case.use_case_id {
            return Ok(InsertBatch::skip());
        }
    }
```

1) Only pre-parse the message if you need to (there is a killswitch)
2) Rust is a lot more powerful in pattern matching and combinators, which means you basically never have to `unwrap()`